### ISSUE TITLE
[BUG]: Ray System Error

### ISSUE THREAD

**Issue by turkalpmd** [pycaret/pycaret#3639]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I haven't been using Pycaret for about a month because I've been on vacation, but I checked and noticed that three new versions have been released. Great job, you're doing an excellent job. I immediately updated my version. Then I tried running my previously working model. As soon as I reached the "Tune" section, I encountered a serious problem. I visited GPT, Bard, Stack Overflow, as well as your discussion and issues sections, but couldn't find a solution.

First, Ray gave an "out of memory" error, which is impossible since I'm working on a computer with 32 GB of RAM, an i7 CPU with 24 cores, and an RTX 3090 GPU, with only 2 MB of data. Then I used pip install to update Ray and also updated the grpcio library that was causing the error. I even uninstalled and reinstalled Pycaret 4-5 times to check if it became incompatible. But each time, I started getting an error like 'Broken pipe.' I don't know what to do to solve this problem.

CPU:
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         46 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  24
  On-line CPU(s) list:   0-23
Vendor ID:               GenuineIntel
  Model name:            13th Gen Intel(R) Core(TM) i7-13700K
    CPU family:          6
    Model:               183
    Thread(s) per core:  2
    Core(s) per socket:  16
    Socket(s):           1
    Stepping:            1
    CPU max MHz:         5400,0000
    CPU min MHz:         800,0000

RAM:
               total        used        free      shared  buff/cache   available
Mem:            31Gi       5,8Gi        16Gi       179Mi       8,9Gi        24Gi
Swap:          2,0Gi          0B       2,0Gi

GPU:
| NVIDIA-SMI 525.125.06   Driver Version: 525.125.06   CUDA Version: 12.0     |
|  0%   50C    P8    36W / 370W |    909MiB / 24576MiB |      3%      Default |

### Reproducible Example

```python
tuned_model = tune_model(
                            estimator=model,
                            n_iter=30,
                            optimize='AUC',
                            search_library='tune-sklearn',#tune-sklearn, optuna
                            search_algorithm='bayesian',# tpe | random | grid | bayesian  | hyperopt | optuna | bohb
                            early_stopping='Hyperband', # asha  | Hyperband | median
                            early_stopping_max_iters=10,
                            choose_better=True, 
                            return_tuner=False                           
                           )
predict = predict_model(tuned_model,test)     
```
```


### Expected Behavior

.
.
.
`
(_Trainable pid=253128) [LightGBM] [Warning] bagging_freq is set=7, subsample_freq=0 will be ignored. Current value: bagging_freq=7
(_Trainable pid=253128) [LightGBM] [Warning] bagging_fraction is set=0.6342100786877556, subsample=1.0 will be ignored. Current value: bagging_fraction=0.6342100786877556
`
.
.
.

### Actual Results

```python-traceback
here is the 3.0.3 Error

---------------------------------------------------------------------------
RaySystemError                            Traceback (most recent call last)
/tmp/ipykernel_60394/232500569.py in ()
----> 1 tuned_model = tune_model(
      2                             estimator=model,
      3                             n_iter=30,
      4                             optimize='AUC',
      5                             search_library='tune-sklearn',#tune-sklearn, optuna

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/utils/generic.py in wrapper(*args, **kwargs)
    963                 if globals_d[name] is None:
    964                     raise ValueError(message)
--> 965             return func(*args, **kwargs)
    966 
    967         return wrapper

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/classification/functional.py in tune_model(estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, return_train_score, **kwargs)
   1214     """
   1215 
-> 1216     return _CURRENT_EXPERIMENT.tune_model(
   1217         estimator=estimator,
   1218         fold=fold,

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/classification/oop.py in tune_model(self, estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, return_train_score, **kwargs)
   1566         """
   1567 
-> 1568         return super().tune_model(
   1569             estimator=estimator,
   1570             fold=fold,

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/internal/pycaret_experiment/supervised_experiment.py in tune_model(self, estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, return_train_score, **kwargs)
   2653                         model_grid.fit(data_X, data_y, groups=groups, **fit_kwargs)
   2654             else:
-> 2655                 model_grid.fit(data_X, data_y, groups=groups, **fit_kwargs)
   2656             best_params = model_grid.best_params_
   2657             self.logger.info(f"best_params: {best_params}")

~/anaconda3/envs/thesis/lib/python3.8/site-packages/tune_sklearn/tune_basesearch.py in fit(self, X, y, groups, tune_params, **fit_params)
    625 
    626         """
--> 627         return self._fit(X, y, groups, tune_params, **fit_params)
    628 
    629     def score(self, X, y=None):

~/anaconda3/envs/thesis/lib/python3.8/site-packages/tune_sklearn/tune_basesearch.py in _fit(self, X, y, groups, tune_params, **fit_params)
    531 
    532         self._fill_config_hyperparam(config)
--> 533         self.analysis_ = self._tune_run(X, y, config, resources_per_trial,
    534                                         tune_params, fit_params)
    535 

~/anaconda3/envs/thesis/lib/python3.8/site-packages/tune_sklearn/tune_search.py in _tune_run(self, X, y, config, resources_per_trial, tune_params, fit_params)
    809             run_args, tune_params)
    810 
--> 811         trainable = tune.with_parameters(
    812             trainable,
    813             X=X,

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/tune/trainable/util.py in with_parameters(trainable, **kwargs)
    291     prefix = f"{str(trainable)}_"
    292     for k, v in kwargs.items():
--> 293         parameter_registry.put(prefix + k, v)
    294 
    295     trainable_name = getattr(trainable, "__name__", "tune_with_parameters")

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/tune/registry.py in put(self, k, v)
    294         self.to_flush[k] = v
    295         if ray.is_initialized():
--> 296             self.flush()
    297 
    298     def get(self, k):

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/tune/registry.py in flush(self)
    306                 self.references[k] = v
    307             else:
--> 308                 self.references[k] = ray.put(v)
    309         self.to_flush.clear()

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/_private/auto_init_hook.py in auto_init_wrapper(*args, **kwargs)
     16     def auto_init_wrapper(*args, **kwargs):
     17         auto_init_ray()
---> 18         return fn(*args, **kwargs)
     19 
     20     return auto_init_wrapper

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/_private/client_mode_hook.py in wrapper(*args, **kwargs)
    101             if func.__name__ != "init" or is_client_mode_enabled_by_default:
    102                 return getattr(ray, func.__name__)(*args, **kwargs)
--> 103         return func(*args, **kwargs)
    104 
    105     return wrapper

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/_private/worker.py in put(value, _owner)
   2610     with profiling.profile("ray.put"):
   2611         try:
-> 2612             object_ref = worker.put_object(value, owner_address=serialize_owner_address)
   2613         except ObjectStoreFullError:
   2614             logger.info(

~/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/_private/worker.py in put_object(self, value, object_ref, owner_address)
    708         # reference counter.
    709         return ray.ObjectRef(
--> 710             self.core_worker.put_serialized_object_and_increment_local_ref(
    711                 serialized_value, object_ref=object_ref, owner_address=owner_address
    712             ),

python/ray/_raylet.pyx in ray._raylet.CoreWorker.put_serialized_object_and_increment_local_ref()

python/ray/_raylet.pyx in ray._raylet.CoreWorker._create_put_buffer()

python/ray/_raylet.pyx in ray._raylet.check_status()

RaySystemError: System error: Broken pipe
```

Here is the 3.0.4 Error;

```
2023-07-17 01:56:25,978	ERROR services.py:1207 -- Failed to start the dashboard , return code 1
2023-07-17 01:56:25,979	ERROR services.py:1232 -- Error should be written to 'dashboard.log' or 'dashboard.err'. We are printing the last 20 lines for you. See 'https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure' to find where the log file is.
2023-07-17 01:56:25,979	ERROR services.py:1276 -- 
The last 20 lines of /tmp/ray/session_2023-07-17_01-56-24_626228_128921/logs/dashboard.log (it contains the error message from the dashboard): 
    from ray.util.state.common import (
  File "/home/turkalp/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/util/state/__init__.py", line 1, in <module>
    from ray.util.state.api import (
  File "/home/turkalp/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/util/state/api.py", line 17, in <module>
    from ray.util.state.common import (
  File "/home/turkalp/anaconda3/envs/thesis/lib/python3.8/site-packages/ray/util/state/common.py", line 411, in <module>
    class ActorState(StateSchema):
  File "pydantic/dataclasses.py", line 224, in pydantic.dataclasses.dataclass.wrap
  File "pydantic/dataclasses.py", line 347, in pydantic.dataclasses._add_pydantic_validation_attributes
  File "pydantic/dataclasses.py", line 400, in pydantic.dataclasses.create_pydantic_model_from_dataclass
  File "pydantic/main.py", line 1026, in pydantic.main.create_model
  File "pydantic/main.py", line 197, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/fields.py", line 506, in pydantic.fields.ModelField.infer
  File "pydantic/fields.py", line 436, in pydantic.fields.ModelField.__init__
  File "pydantic/fields.py", line 552, in pydantic.fields.ModelField.prepare
  File "pydantic/fields.py", line 668, in pydantic.fields.ModelField._type_analysis
  File "/home/turkalp/anaconda3/envs/thesis/lib/python3.8/typing.py", line 774, in __subclasscheck__
    return issubclass(cls, self.__origin__)
TypeError: issubclass() arg 1 must be a class
2023-07-17 01:56:26,011	INFO worker.py:1636 -- Started a local Ray instance.
(raylet) [2023-07-17 01:56:26,421 E 129607 129657] (raylet) agent_manager.cc:135: The raylet exited immediately because the Ray agent failed. The raylet fate shares with the agent. This can happen because the Ray agent was unexpectedly killed or failed. Agent can fail when
(raylet) - The version of `grpcio` doesn't follow Ray's requirement. Agent can segfault with the incorrect `grpcio` version. Check the grpcio version `pip freeze | grep grpcio`.
(raylet) - The agent failed to start because of unexpected error or port conflict. Read the log `cat /tmp/ray/session_latest/logs/dashboard_agent.log`. You can find the log file structure here https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure.
(raylet) - The agent is killed by the OS (e.g., out of memory).
[2023-07-17 01:56:26,615 E 128921 129677] core_worker.cc:572: :info_message: Attempting to recover 2 lost objects by resubmitting their tasks. To disable object reconstruction, set @ray.remote(max_retries=0).
---------------------------------------------------------------------------
RaySystemError                            Traceback (most recent call last)
/tmp/ipykernel_128921/232500569.py in ()
----> 1 tuned_model = tune_model(
      2                             estimator=model,
      3                             n_iter=30,
      4                             optimize='AUC',
      5                             search_library='tune-sklearn',#tune-sklearn, optuna

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/utils/generic.py in wrapper(*args, **kwargs)
    963                 if globals_d[name] is None:
    964                     raise ValueError(message)
--> 965             return func(*args, **kwargs)
    966 
    967         return wrapper

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/classification/functional.py in tune_model(estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, return_train_score, **kwargs)
   1214     """
   1215 
-> 1216     return _CURRENT_EXPERIMENT.tune_model(
   1217         estimator=estimator,
   1218         fold=fold,

~/anaconda3/envs/thesis/lib/python3.8/site-packages/pycaret/classification/oop.py in tune_model(self, estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, return_train_score, **kwargs)
   1562         """
...
python/ray/_raylet.pyx in ray._raylet.CoreWorker._create_put_buffer()

python/ray/_raylet.pyx in ray._raylet.check_status()

RaySystemError: System error: Broken pipe

```
```


### Installed Versions

<details>
System:
    python: 3.8.16 (default, Jun 12 2023, 18:09:05)  [GCC 11.2.0]
executable: /home/turkalp/anaconda3/envs/thesis/bin/python
   machine: Linux-5.19.0-46-generic-x86_64-with-glibc2.17

PyCaret required dependencies:
                 pip: 23.2
          setuptools: 67.8.0
             pycaret: 3.0.4
             IPython: 7.34.0
          ipywidgets: 7.7.5
                tqdm: 4.65.0
               numpy: 1.23.0
              pandas: 1.5.3
              jinja2: 3.1.2
               scipy: 1.10.1
              joblib: 1.3.1
             sklearn: 1.2.2
                pyod: 1.1.0
            imblearn: 0.10.1
   category_encoders: 2.6.1
            lightgbm: 3.3.5
               numba: 0.57.1
            requests: 2.31.0
          matplotlib: 3.7.1
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.15.0
    plotly-resampler: Not installed
             kaleido: 0.2.1
           schemdraw: 0.15
         statsmodels: 0.14.0
              sktime: 0.17.0
               tbats: 1.1.3
            pmdarima: 2.0.3
              psutil: 5.9.5
          markupsafe: 2.1.3
             pickle5: Not installed
         cloudpickle: 2.2.1
         deprecation: 2.1.0
              xxhash: 3.2.0
           wurlitzer: 3.0.3

PyCaret optional dependencies:
                shap: 0.41.0
           interpret: 0.4.2
                umap: 0.5.3
    pandas_profiling: 4.3.1
  explainerdashboard: 0.4.2.2
             autoviz: 0.1.720
           fairlearn: 0.7.0
          deepchecks: 0.17.3
             xgboost: 1.7.6
            catboost: 1.2
              kmodes: 0.12.2
             mlxtend: 0.22.0
       statsforecast: 1.5.0
        tune_sklearn: 0.4.6
                 ray: 2.5.1
            hyperopt: 0.2.7
              optuna: 3.2.0
               skopt: 0.9.0
              mlflow: 1.30.0
              gradio: 3.35.2
             fastapi: 0.97.0
             uvicorn: 0.18.3
              m2cgen: 0.10.0
           evidently: 0.2.8
               fugue: 0.8.5
           streamlit: Not installed
             prophet: Not installed

</details>

**Comment by turkalpmd** [comment]
Hello,

I managed to resolve the issue on my own, so I'd like to share my findings about Pycaret requiring Python 3.10 to run properly. In the previous version (3.8), there were conflicts and errors with ray, which seem to have been resolved in Python 3.10.

I wanted to share this information in case someone else encounters a similar problem and needs some guidance.

Thank you!

**Comment by turkalpmd** [comment]
Hello,

I managed to resolve the issue on my own, so I'd like to share my findings about Pycaret requiring Python 3.10 to run properly. In the previous version (3.8), there were conflicts and errors with ray, which seem to have been resolved in Python 3.10.

I wanted to share this information in case someone else encounters a similar problem and needs some guidance.

Thank you!
